src/javascript/binary/socket.js

```diff
@@ -62,16 +62,18 @@ var BinarySocket = (function () {
             if(!data.hasOwnProperty('passthrough')){
                 data.passthrough = {};
             }
-            data.passthrough.req_number = ++req_number;
-            timeouts[req_number] = setInterval(function(){
-                if(typeof reloadPage === 'function'){
-                    var r = confirm("The server didn't respond. Reload page?");
-                    console.log(data,'Last Error Request');
-                    if (r === true) {
-                        reloadPage();
-                    } 
-                }
-            }, 7*1000);
+            if(data.contracts_for || data.proposal){
+                data.passthrough.req_number = ++req_number;
+                timeouts[req_number] = setInterval(function(){
+                    if(typeof reloadPage === 'function'){
+                        var r = confirm("The server didn't respond to the request:\n\n"+JSON.stringify(data)+"\n\nReload page?");
+                        if (r === true) {
+                            reloadPage();
+                        } 
+                    }
+                }, 7*1000);
```

Comment:
I agree with 1)
I agree with 2) but we need to be even more aggressive, and then delay the timeout. We can have a notice for 'offline' work.

We reverted the Trade page to the old one, and there might be reworkings on the back end too.
If this will take time, we can just replace this with binary-live-api that will handle this.